### DISCUSSION TITLE
The ```LowerConvsToMatMul``` function does not seem to support the ordinary im2col transformation of grouped convolutions.

### DISCUSSION THREAD

**Discussion by sansi-zhang** [Xilinx/finn#883] [Q&A]
@maltanar 
Hi, I got a "reshape" error while using LowerConvsToMatMul to process grouped convolutions.

I set the number of groups to be a factor of the number of input channels, and I've seen the corresponding value being written to the node in onnx, but there's an error when using LowerConvsToMatMul, it seems that it doesn't correctly recognize grouped convolutions with fewer groups than the number of input channels.

Can I fix this by adjusting LowerConvsToMatMul?

If can I hope you can give me some guidance and advice.

Thanks.

**Comment by fpjentzsch** [comment]
Hi,
FINN does indeed not support arbitrarily grouped convolutions. Only normal convolutions (implemented by ConvolutionInputGenerator + MVAU) and depthwise convolutions (groups = channels, implemented by ConvolutionInputGenerator + VVAU).

Supporting this would require more than simply adjusting LowerConvsToMatMul. The ConvolutionInputGenerator(_rtl) and MVAU or VVAU would need to be extended as well. I'm not aware of any plans to implement this, but we are always open for pull requests :-)

**Reply by nickfraser** [reply]
Just a quick note to be explicit, if `groups>1`, `LowerConvsToMatMul` fails in all cases except when `groups == in_channels == out_channels`. I've open an issue about this meanwhile, #918.
